plugins/permission-node/src/ServerPermissionClient.ts

```diff
@@ -62,58 +68,92 @@ export class ServerPermissionClient implements PermissionEvaluator {
-    this.tokenManager = options.tokenManager;
-    this.permissionEnabled = options.permissionEnabled;
+    this.#auth = options.auth;
+    this.#permissionClient = options.permissionClient;
+    this.#permissionEnabled = options.permissionEnabled;
   }
 
   async authorizeConditional(
     queries: QueryPermissionRequest[],
-    options?: EvaluatorRequestOptions,
+    options?: PermissionsServiceRequestOptions,
   ): Promise<PolicyDecision[]> {
-    return (await this.isEnabled(options?.token))
-      ? this.permissionClient.authorizeConditional(queries, options)
-      : queries.map(_ => ({ result: AuthorizeResult.ALLOW }));
+    if (await this.#shouldPermissionsBeApplied(options)) {
+      return this.#permissionClient.authorizeConditional(
+        queries,
+        await this.#getRequestOptions(options),
+      );
+    }
+
+    return queries.map(_ => ({ result: AuthorizeResult.ALLOW }));
   }
 
   async authorize(
     requests: AuthorizePermissionRequest[],
-    options?: EvaluatorRequestOptions,
+    options?: PermissionsServiceRequestOptions,
   ): Promise<AuthorizePermissionResponse[]> {
-    return (await this.isEnabled(options?.token))
-      ? this.permissionClient.authorize(requests, options)
-      : requests.map(_ => ({ result: AuthorizeResult.ALLOW }));
+    if (await this.#shouldPermissionsBeApplied(options)) {
+      return this.#permissionClient.authorize(
+        requests,
+        await this.#getRequestOptions(options),
+      );
+    }
+
+    return requests.map(_ => ({ result: AuthorizeResult.ALLOW }));
   }
 
-  private async isValidServerToken(
-    token: string | undefined,
-  ): Promise<boolean> {
-    if (!token) {
-      return false;
+  async #getRequestOptions(options?: PermissionsServiceRequestOptions) {
+    if (options && 'credentials' in options) {
+      if (this.#auth.isPrincipal(options.credentials, 'none')) {
+        return {};
+      }
+
+      return this.#auth.getPluginRequestToken({
+        onBehalfOf: options.credentials,
+        targetPluginId: 'permissions',
+      });
     }
-    return this.tokenManager
-      .authenticate(token)
-      .then(() => true)
-      .catch(() => false);
+
+    return options;
   }
 
-  private async isEnabled(token?: string) {
-    // Check if permissions are enabled before validating the server token. That
-    // way when permissions are disabled, the noop token manager can be used
-    // without fouling up the logic inside the ServerPermissionClient, because
-    // the code path won't be reached.
-    return this.permissionEnabled && !(await this.isValidServerToken(token));
+  async #shouldPermissionsBeApplied(
+    options?: PermissionsServiceRequestOptions,
+  ) {
+    if (!this.#permissionEnabled) {
+      return false;
+    }
+
+    let credentials: BackstageCredentials;
+    if (options && 'credentials' in options) {
+      credentials = options.credentials;
+    } else {
+      if (!options?.token) {
+        return true;
+      }
```

Comment:
Minor nit: structuring the conditionals this way, we lose the semantic `isValidServerToken` var name, which I think was very helpful in terms of readability (which is arguably more important now that we have multiple levels of nesting)

Reply:
hmm, not sure I follow how you'd want to structure it tbh? the server token check is now down in the `isPrincipal` call, which feels straight-forward enough at this point